[MM-20058] Team and Channel roles in group configuration screen

hahmadia opened the pull request:
#### Summary
<!--
A description of what this pull request does.
-->
Displays roles for teams and channels in the group configuration screen

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->
https://mattermost.atlassian.net/browse/MM-20058

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
Awaiting on https://github.com/mattermost/mattermost-server/pull/13361
Awaiting on https://github.com/mattermost/mattermost-redux/pull/1009 

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->
![image](https://user-images.githubusercontent.com/17804942/71600235-159e5c80-2b1c-11ea-8e8a-288c933290ea.png)

hahmadia commented:
@asaadmahmood This is the PR with the CSS fixes. You will need to go to System Console > Groups (Beta) > Pick a Group > Team and Channel Memberships. Add a couple teams and channels. I'm trying to align everything under the right header. Here is my current progress but when assigned roles changes or the type has extra text like `Team (Private)` or `Channel (Private)`, it messes the alignment and I'm stuck on what to do. Hoping you can help out with this. Thank you!!  
![image](https://user-images.githubusercontent.com/17804942/71392663-e1381700-25d6-11ea-9192-1bea719d53ad.png)

asaadmahmood commented:
Pushed a fix, have a look.

hahmadia commented:
**TODO:** Add Cypress Tests

mkraft commented on components/widgets/admin_console/admin_panel.tsx:
```diff
@@ -37,13 +37,13 @@ const AdminPanel: React.FC<Props> = (props: Props) => (
                     />
                 </h3>
-                <span>
+                <div className='margin-top'>
```
Maybe we could change to a name that is more idiomatic that doesn't describe the styling.

asaadmahmood replied:
We should most definitely, bootstrap 4 uses `mt-x` etc, we can use that.
But that can be a separate ticket since these classes are used elsewhere aswell.

srkgupta commented on e2e/cypress/integration/system_console/groups_assign_roles_spec.js:
```diff
@@ -0,0 +1,58 @@
+            for (let i = 1; i < el[0].childNodes[1].rows.length; i++) {
+                name = el[0].childNodes[1].rows[i].cells[0].innerText;
+                cy.wrap(el[0].childNodes[1].rows[i]).get(`#${name}_current_role`).click();
+                cy.wrap(el[0].childNodes[1].rows[i]).get(`#${name}_role_to_be`).click();
```
Before clicking on any element, please add a statement to scrollIntoView. Something like this:

```cy.wrap(el[0].childNodes[1].rows[i]).get(`#${name}_role_to_be`).scrollIntoView().click();```

Should also be updated in the previous line

srkgupta commented on e2e/cypress/integration/system_console/groups_assign_roles_spec.js:
```diff
@@ -0,0 +1,58 @@
+            }
+        });
+    });
+});
```
Since this is an enterprise level feature, please move this testcase to ```e2e/cypress/integration/enterprise/system_console``` directory

hahmadia commented:
@srkgupta 
Hey! I addressed all the problems you mentioned and I added functions to remove all the teams and channels associated to that group prior to adding my own.

srkgupta commented on e2e/cypress/integration/enterprise/system_console/groups_assign_roles_spec.js:
```diff
@@ -0,0 +1,110 @@
+                    getChannelsAssociatedToGroupAndUnlink(groupId);
+                    cy.get('#developers_edit').click();
+                });
+            } else {
```
Can you also add a step to click on "Not Linked" link, if it's present. This way, it will also handle the case where the group wasn't linked before. You can do this by adding the following step in the else condition before clicking on configure button. 

```
                // # Link the Group if its not linked before
                if (el.find('.icon.fa-unlink').length > 0) {
                    el.find('.icon.fa-unlink').click();
                }
```

srkgupta commented on e2e/cypress/integration/enterprise/system_console/groups_assign_roles_spec.js:
```diff
@@ -0,0 +1,110 @@
+        cy.get('#multiSelectList').first().click();
+        cy.get('#saveItems').click();
+
+        // # Add the first team in the group list then save
```
The comment should be "Add the first channel in the group list then save"

mattermod commented:
Will try to auto merge this PR once all tests and checks are passing. This might take up to an hour.
